src/net_processing.cpp

```diff
@@ -1920,27 +1939,27 @@ void PeerManagerImpl::ProcessHeadersMessage(CNode& pfrom, const Peer& peer,
         // because it is set in UpdateBlockAvailability. Some nullptr checks
         // are still present, however, as belt-and-suspenders.
 
-        if (received_new_header && pindexLast->nChainWork > ::ChainActive().Tip()->nChainWork) {
+        if (received_new_header && pindexLast->nChainWork > m_chainman.ActiveChain().Tip()->nChainWork) {
```

Comment:
This sequence of `m_chainman.ActiveChain()` function calls is repeatedly taking a `cs_main` lock and releasing it again (while holding `cs_main` the whole time). That's fine, and correct, but perhaps it'd be better to just take a local `const CChainstate&` variable once to avoid the multiple function calls.